test/AutoDiff/Sema/DerivedConformances/class_differentiable.swift

```diff
@@ -109,28 +109,13 @@ final class VectorSpacesEqualSelf: Differentiable & DummyAdditiveArithmetic {
   var w: Float
   var b: Float
   typealias TangentVector = VectorSpacesEqualSelf
+  var zeroTangentVectorInitializer: () -> TangentVector { fatalError() }
```

Comment:
Shouldn't these use the default implementation (`.zero`)?  Is it because derivation won't be triggered if you provide a default implementation?

Reply:
> Is it because derivation won't be triggered if you provide a default implementation?

Yes, we have to remove the dummy default implementation of `Differentiable.zeroTangentVectorInitializer` using `TangentVector.zero` because its existence prevents derived conformances from triggering.

This requires "primitive" `Differentiable`-conforming types to implement `zeroTangentVectorInitializer`.